students/TacSPx/lesson04/mailroompart2.py

```diff
@@ -0,0 +1,216 @@
+# ---------------------------------------------------------------------------- #
+# imports
+import sys
+from collections import defaultdict
+from operator import itemgetter
+
+# Data ---------------------------------------------------------------------- #
+# Task 1: Create data
+# Create a data structure that holds a list of your donors and a history of the amounts they have donated.
+# Have least five donors, with between 1 and 3 donations each, store in the global namespace.
+
+
+donor_info = {"Jim Zorn": [3772.32, 1201.17],
+              "Jermaine Kearse": [877.33],
+              "Marcus Trufant": [1563.23, 1043.87, 1.32],
+              "K.J. Wright": [21663.23, 300.87, 100432.0],
+              "Curt Warner": [663.23, 300.87, 10432.0],
+              }
+
+
+# Processing  --------------------------------------------------------------- #
+# Task 2:  Menu choices
+# Prompt the user to choose from a menu of 3 actions: “Send a Thank You”, “Create a Report” or “quit”.
+
+###############################################
+#   "Menu of Choices"
+###############################################
+def main():
+    '''
+    Menu of Choices:  Thank You, Report, Create a Report
+    :return:
+    '''
+    switch_choice_dict = {"1": thank_you, "2": report, "3": all_thank_you, "4": exit_program}
+    while True:
+        print("""
+        \n**** XYZ Nonprofit Charity ****\n
+        Menu of Options
+        1) Send a "Thank You"
+        2) Create a Report
+        3) Send letters to all Donors
+        4) Exit Program
+        """)
+        # User menu input
+        str_choice = str(input("Which option would you like to perform? [1 to 4] - "))
+        print()  # added a new line for spacing looks
+        # dict switch choice, if not a menu choice then run non_selection function
+        switch_choice_dict.get(str_choice, non_selection)()
+
+
+###############################################
+#   #1 Send a "Thank You"
+###############################################
+def thank_you():
+    '''
+    Thank You:  Will append donations to existing donors, or adds new donors with donations.
+    User can see list of existing donors by typing 'list'.  Once donation is entered script will create
+    custom thank you letter/ email.
+    :return:
+    '''
+    while True:
+        donor_name = input("Type 'List' to see all names in Database or\nType First and Last Name: ").title()
+        if (donor_name.lower() == 'list'):
+            print("\nDONOR REGISTRY:")
+            donor_sort = (sorted(donor_info.keys()))  # used .keys() to sort on key
+            for key in donor_sort:
+                print(key)
+            input("Press Enter to Continue")
+            print()
+            continue
+
+        if (donor_name.lower() != 'list'):
+            donor_amount = int(input("Enter Amount of Donation: $"))
+            # if name is in dict append dollar amount to donors bucket
+            for i in donor_info:
```

Comment:
looping over a dictionary to find the key is incorrect - you can access and check they key by either index access or via get() method.